src/common/markdown.ts

```diff
@@ -43,16 +43,49 @@ const defaultTextRender = markdown.renderer.rules.text;
 
 type MarkdownMention = Pick<User, 'id' | 'username'>;
 
+export const mentionSpecialCharacters = new RegExp('[^a-zA-Z0-9_@]', 'g');
+
+type ReplacedCharacters = string[];
+
+const getReplacedCharacters = (word: string): [string, ReplacedCharacters] => {
+  const specialCharacters = [];
+  word.split('').forEach((char) => {
+    const matched = char.match(mentionSpecialCharacters);
+    if (matched?.[0]?.length) {
+      specialCharacters.push(char);
+    }
+  });
```

Comment:
You can use something like this: https://stackoverflow.com/a/2295681/2650104
Instead of splitting every character and doing a regex match operation